test/e2e/runner/main.go

```diff
@@ -52,6 +52,9 @@ func NewCLI() *CLI {
 			if err := Cleanup(cli.testnet); err != nil {
 				return err
 			}
+			if !cli.preserve {
+				defer Cleanup(cli.testnet)
```

Comment:
I suggest not deferring up front, but leaving this until the end (where the original code was) so we can log, e.g.,

```go
	if cli.preserve {
		log.Print("Preserving testnet contents because -preserve=true")
	} else if err := Cleanup(cli.testnet); err != nil {
		return err
	}
```

Just so the operator (us) can tell that this happened from the log.

Reply:
I think we can still log and keep the defer? The main goal here just being that the contents of the data dirs / docker and the docker networks/containers are left around. For example, if `Wait` fails because a node never enters a good state, then we preserve all of the test and I'm not yet sure I understand why we'd want to do that if preserve wasn't specified.

Reply:
Sure, I'm not opposed to the deferral, I mainly figured it would be nice to have it log what it's doing.